galsim/sed.py

```diff
@@ -292,47 +489,58 @@ def __add__(self, other):
-        wave_list = wave_list[wave_list <= red_limit]
-        wave_list = wave_list[wave_list >= blue_limit]
+        if self.dimensionless and other.dimensionless:
+            flux_type = '1'
+            _spectral = False
+            _dimensionless = True
+        elif self.spectral and other.spectral:
+            flux_type = 'fphotons'
+            _spectral = True
+            _dimensionless = False
+        else:
+            raise TypeError("Cannot add SEDs with incompatible dimensions.")
 
-        return SED(spec, wave_type='nm', flux_type='fphotons',
+        spec = lambda w: self(w*(1.0+self.redshift)) + other(w*(1.0+self.redshift))
+        wave_list, blue_limit, red_limit = galsim.utilities.combine_wave_list([self, other])
+
+        return SED(spec, wave_type='nm', flux_type=flux_type,
                    redshift=self.redshift, _wave_list=wave_list,
-                   _blue_limit=blue_limit, _red_limit=red_limit)
+                   _blue_limit=blue_limit, _red_limit=red_limit,
+                   _spectral=_spectral, _dimensionless=_dimensionless)
 
     def __sub__(self, other):
         # Subtract two SEDs, with the same caveats as adding two SEDs.
         return self.__add__(-1.0 * other)
 
-    def copy(self):
-        import copy
-        return copy.deepcopy(self)
-
     def withFluxDensity(self, target_flux_density, wavelength):
         """ Return a new SED with flux density set to `target_flux_density` at wavelength
-        `wavelength`.  Note that this normalization is *relative* to the `flux` attribute of the
-        chromaticized GSObject.
+        `wavelength`.
 
-        @param target_flux_density  The target *relative* normalization in photons / nm.
-        @param wavelength           The wavelength, in nm, at which flux density will be set.
+        @param target_flux_density  The target normalization in photons/nm/cm^2/s.
+        @param wavelength           The wavelength, in nm, at which the flux density will be set.
 
         @returns the new normalized SED.
         """
+        if self.dimensionless:
+            raise TypeError("Cannot set flux density of dimensionless SED.")
+        if isinstance(wavelength, units.Quantity):
+            wavelength_nm = wavelength.to(units.nm, units.spectral())
         current_flux_density = self(wavelength)
+        if isinstance(target_flux_density, units.Quantity):
+            target_flux_density = target_flux_density.to(
+                    _photons, units.spectral_density(wavelength_nm)).value
         factor = target_flux_density / current_flux_density
         return self * factor
 
     def withFlux(self, target_flux, bandpass):
-        """ Return a new SED with flux through the Bandpass `bandpass` set to `target_flux`. Note
+        """ Return a new SED with flux through the Bandpass `bandpass` set to `target_flux`.  Note
         that this normalization is *relative* to the `flux` attribute of the chromaticized GSObject.
```

Comment:
I think this is mostly a remnant from the previous version where a `Chromatic` took its normalization (in `evaluateAtWavelength` and `drawImage`) from both `chrom.obj` and `chrom.SED`.  I replaced the docstring here, in `withFluxDensity`, and in `withMagnitude` with the equivalent of:

```
        Return a new SED with flux through the Bandpass `bandpass` set to `target_flux`.  See
        ChromaticObject docstring for information about how SED normalization affects
        ChromaticObject normalization.
```